### DISCUSSION TITLE
Behaviour on keyboard return key

### DISCUSSION THREAD

**Discussion by wtpalexander** [mobilinked/MbSwiftUIFirstResponder#5] [Q&A] [answered]
Hi! Thanks for this project, its really helpful! Is the intended behaviour that the first responder State field would update when hitting 'return' on the keyboard as well? For some reason it only seems to update when I tap away from the field rather than after tapping the return key on the keyboard.

**Comment by mobilinked** [comment]
Thank you for the comments. 

The Mac version supports to resign the first responder by click outside and keyboard events like return, ESC, and tab keys if suitable. 
The iOS version only supports to resign by tap outside (you can check the API to see resignable events enum)

I will add keyboard support to the iOS version when I have time for it (and if it's possible). 

------
This solution is a temp solution as Apple does not provide one now. This solution tried to analyze the backed NSView and UIView and add some more controls. We are trying to make the API simple and make it easy to migrate when Apple provides an API future.

**Reply by wtpalexander** [reply]
Sure thing, I understand its a temporary cover whilst we wait for Apple to expand the SwiftUI framework but its an effective fix in the meantime, thank you!

**Comment by mobilinked** [comment, marked as answer]
And I'd like to suggest a simple solution for your case:

```swift
TextField(
    "xxx", 
    text: $xxx,
    onCommit: {
        DispatchQueue.main.async {
           self.firstResponder = nil
        }
    }
)
```

**Reply by wtpalexander** [reply]
Ah great, thank you!
